src/search.rs

```diff
@@ -176,27 +184,38 @@ fn find_max(
 pub fn viterbi_search<D: Data<Elem = f32>>(
     network_output: &ArrayBase<D, Ix2>,
     alphabet: &[String],
+    qstring: bool,
+    qscale: f32,
+    qbias: f32,
 ) -> Result<(String, Vec<usize>), SearchError> {
     assert!(!alphabet.is_empty());
     assert!(!network_output.is_empty());
     assert_eq!(alphabet.len(), network_output.shape()[1]);
 
     let mut path = Vec::new();
+    let mut quality = String::new();
     let mut sequence = String::new();
 
     let mut last_label = None;
     for (idx, pr) in network_output.outer_iter().enumerate() {
-        let (label, _) = Zip::indexed(pr)
+        let (label, prob) = Zip::indexed(pr)
             .fold_while(None, find_max)
             .into_inner()
             .unwrap(); // only an empty network_output could give us None
         if label != 0 && last_label != Some(label) {
             sequence.push_str(&alphabet[label]);
             path.push(idx);
+            if qstring {
+                quality.push(phred(prob, qscale, qbias));
```

Comment:
When a label spans multiple entries in `network_output`, the quality score will be the probability of the first entry, and the other entries will be ignored. Is that the right calculation?

Reply:
The mean across the span would probably be better.